### ISSUE TITLE
Package cleanup rules not working

### ISSUE THREAD

**Issue by denyskon** [go-gitea/gitea#32349]
### Description

I have package cleanup rules set up for container, generic and debian packages.
None of them work, and this error message appears in admin notices:
```
Cron: Cleanup expired packages canceled: CleanupRule [2]: debian.BuildAllRepositoryFiles failed: failed to build repository files [all/main/amd64]: package blob does not exist
```

### Gitea Version

1.23.0+dev-587-g7cf611d197 (nightly)

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

Official nightly docker image

### Database

SQLite

**Comment by wxiaoguang** [comment]
Reproduced:

![image](https://github.com/user-attachments/assets/f2e7c9b2-40e9-46c3-8c22-866a2d501595)

**Comment by wxiaoguang** [comment]
->  Fix db engine #32351
